diff --git a/Makefile.orig b/Makefile
similarity index 66%
rename from Makefile.orig
rename to Makefile
index eb9c0b0..382af98 100644
--- a/Makefile.orig
+++ b/Makefile
@@ -1,3 +1,5 @@
+.PHONY: clean cleanall
+
 CC = g++
 
 calc: calc.o main.o parser.o lexer.o
@@ -15,3 +17,9 @@ lexer.cpp:lexer.l parser.hpp
 	flex -o lexer.cpp lexer.l
 	
 lexer.o:lexer.cpp
+
+clean:
+	rm -f *.o *.obj
+
+cleanall: clean
+	rm -f parser.cpp parser.hpp lexer.cpp lexer.hpp calc calc.exe
diff --git a/main.cpp b/main.cpp
index 925d8c4..ea59565 100644
--- a/main.cpp
+++ b/main.cpp
@@ -3,6 +3,12 @@
 #include <cstdio>
 #include <cstdlib>
 #include <new>
+extern "C" {
+  int yylex();
+}
+
+int yyparse(void);
+
 
 void no_memory(void)
 {
